gui/jails/forms.py

```diff
@@ -405,12 +416,28 @@ def save(self):
                     jail_flags |= WARDEN_SET_FLAGS_FLAGS
                     jail_set_args['jflags'] = val
 
-                jail_set_args['flags'] = jail_flags
-                try:
-                    w.set(**jail_set_args)
-                except Exception as e:
-                    self.errors['__all__'] = self.error_class([_(e.message)])
-                    return
+            else:
+                if key == 'jail_mac':
+                    duplicate_mac = 0
+                    while True:
+                        mac_address = generate_randomMAC()
+                        jail_list = Jails.objects.all()
+                        for jail in jail_list:
+                            if jail.jail_mac == mac_address:
+                                duplicate_mac = 1
+                                break
+                        if duplicate_mac != 1:
+                            break
+
+                    jail_flags |= WARDEN_SET_FLAGS_MAC
+                    jail_set_args['mac'] = mac_address
+
+            jail_set_args['flags'] = jail_flags
+            try:
+                w.set(**jail_set_args)
```

Comment:
Here, even if the jail_flags = WARDEN_FLAGS_NONE, it always sets the jail_host in the 'jail_set_args' dictionary. Hence w.set(...) needs to be called in any case.